File: app/src/main/java/com/sjn/taggingplayer/media/provider/multiple/PlaylistProvider.java
package com.sjn.taggingplayer.media.provider.multiple;

import android.content.Context;
import android.support.v4.media.MediaMetadataCompat;

import com.sjn.taggingplayer.R;
import com.sjn.taggingplayer.controller.PlaylistController;
import com.sjn.taggingplayer.utils.MediaIDHelper;

import java.util.List;
import java.util.concurrent.ConcurrentMap;

public class PlaylistProvider extends MultipleListProvider {

    public PlaylistProvider(Context context) {
        super(context);
    }

    @Override
    protected String getMediaKey() {
        return MediaMetadataCompat.METADATA_KEY_GENRE;
    }

    @Override
    protected String getProviderMediaId() {
        return MediaIDHelper.MEDIA_ID_MUSICS_BY_PLAYLIST;
    }

    @Override
    protected int getTitleId() {
        return R.string.media_item_label_playlist;
    }

    @Override
    protected ConcurrentMap<String, List<MediaMetadataCompat>> createTrackListMap(final ConcurrentMap<String, MediaMetadataCompat> musicListById) {
        PlaylistController playlistController = new PlaylistController(mContext);
        return playlistController.getAllPlaylist();
    }

    @Override
    protected int compareMediaList(MediaMetadataCompat lhs, MediaMetadataCompat rhs) {
        return 1;
    }

}

File: app/src/main/java/com/sjn/stamp/media/provider/multiple/PlaylistProvider.kt
package com.sjn.stamp.media.provider.multiple

import android.content.Context
import android.support.v4.media.MediaMetadataCompat

import com.sjn.stamp.R
import com.sjn.stamp.controller.PlaylistController
import com.sjn.stamp.utils.MediaIDHelper

class PlaylistProvider(context: Context) : MultipleListProvider(context) {

    override val mediaKey: String = MediaMetadataCompat.METADATA_KEY_GENRE

    override val providerMediaId: String = MediaIDHelper.MEDIA_ID_MUSICS_BY_PLAYLIST

    override val titleId: Int = R.string.media_item_label_playlist

    override fun updateTrackListMap(musicListById: MutableMap<String, MediaMetadataCompat>) {
        PlaylistController(context).loadAllPlaylist(trackMap)
    }

    override fun compareMediaList(lhs: MediaMetadataCompat, rhs: MediaMetadataCompat): Int =
            compareByTitle(lhs, rhs)

}
